### PR TITLE
AndesBadge to AndesBadgePill (remove modifier)

### PR DESCRIPTION
**Pull request by jperezgonzal**
## Description
- Elimine la interfaz generica AndesBadge y en su lugar utilizo diferentes interfaces públicas.

## Checks
Are you sure you followed all those steps? In such case, let's say with me "I solemnly declare that ...":
- [x] I have coded an example inside the Demo App,
- [ ] I have added proper tests,
- [ ] I have modified the [Changelog](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CHANGELOG.md) file,
- [ ] I have updated GitHub wiki,
- [ ] I have the explicit approval of one or more members of the UX Team,
- [ ] I have updated the version in gradle.properties file.

## Change type
This is easy, let us know what this code is about:
- [ ] This code adds a new component
- [x] This code includes improvements to an existent component
- [ ] This code improves or modifies ONLY the Demo App.

## Check common errors
Whether you are an Android or iOS developer and if you're still there then we'll give you a present:
https://proandroiddev.com/how-to-maximize-androids-ui-reusability-5-common-mistakes-cb2571216a9f

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: components/src/main/java/com/mercadolibre/android/andesui/badge/AndesBadgePill.kt [hunk 1]
```diff
@@ -115,28 +103,26 @@ class AndesBadge : CardView {
      * @param attrs attributes from the XML.
      */
     private fun initAttrs(attrs: AttributeSet?) {
-        andesBadgeAttrs = AndesBadgeAttrsParser.parse(context, attrs)
+        andesBadgeAttrs = AndesBadgeAttrsParser.parsePill(context, attrs)
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/badge/AndesBadgePill.kt:106]: aca en vez de hacer asi, pongamos  `andesBadgePillAttrs = AndesBadgePillAttrsParser.parse(context, attrs)`

File: components/src/main/java/com/mercadolibre/android/andesui/badge/AndesBadgePill.kt [hunk 2]
```diff
@@ -155,7 +141,6 @@ class AndesBadge : CardView {
     /**
      * Creates all the views that are part of this badge.
      * After a view is created then a view id is added to it.
-     *
      */
     private fun initComponents() {
         val container = LayoutInflater.from(context).inflate(R.layout.andes_layout_badge, this)
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/badge/AndesBadgePill.kt:146]: supongo que vamos a usar mas de un layout despues, cambiemosle a andes_layout_badge_pill

File: components/src/main/java/com/mercadolibre/android/andesui/badge/factory/AndesBadgeAttrsParser.kt [hunk 3]
```diff
@@ -5,15 +5,13 @@ import android.util.AttributeSet
...
 import com.mercadolibre.android.andesui.badge.size.AndesBadgeSize
 import com.mercadolibre.android.andesui.badge.type.AndesBadgeType
 
 /**
  * The data class that contains the public components of the badge.
  */
-internal data class AndesBadgeAttrs(
-    val andesBadgeModifier: AndesBadgeModifier,
+internal data class AndesBadgePillAttrs (
     val andesBadgeHierarchy: AndesBadgeHierarchy,
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/badge/factory/AndesBadgeAttrsParser.kt:15]: a las clases propias como hierarchy, size y border podemos ponerle AndesBagdePillHierarchy, etc. Al type no porque es igual para todos.

File: components/src/main/java/com/mercadolibre/android/andesui/badge/factory/AndesBadgeAttrsParser.kt [hunk 4]
```diff
@@ -38,27 +36,20 @@ internal object AndesBadgeAttrsParser {
     private const val ANDES_BADGE_SIZE_LARGE = "3000"
     private const val ANDES_BADGE_SIZE_SMALL = "3001"
 
-    private const val ANDES_BADGE_MODIFIER_PILL = "4000"
+    private const val ANDES_BADGE_BORDER_CORNER = "4000"
+    private const val ANDES_BADGE_BORDER_ROUNDED = "4001"
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/badge/factory/AndesBadgeAttrsParser.kt:40]: Lo mismo con estas constantes, a las que son propias ponerle _PILL_

File: components/src/main/java/com/mercadolibre/android/andesui/badge/factory/AndesBadgeAttrsParser.kt [hunk 5]
```diff
@@ -38,27 +36,20 @@ internal object AndesBadgeAttrsParser {
...
+    private const val ANDES_BADGE_BORDER_STANDARD = "4002"
 
-    private const val ANDES_BADGE_BORDER_CORNER = "5000"
-    private const val ANDES_BADGE_BORDER_ROUNDED = "5001"
-    private const val ANDES_BADGE_BORDER_STANDARD = "5002"
+    fun parsePill(context: Context, attr: AttributeSet?): AndesBadgePillAttrs {
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/badge/factory/AndesBadgeAttrsParser.kt:43]: cambiar por parse solo

File: components/src/main/java/com/mercadolibre/android/andesui/badge/factory/AndesBadgeConfigurationFactory.kt [hunk 6]
```diff
@@ -19,7 +19,7 @@ internal data class AndesBadgeConfiguration(
 
 internal object AndesBadgeConfigurationFactory {
 
-    fun create(context: Context, andesMessageAttrs: AndesBadgeAttrs): AndesBadgeConfiguration {
+    fun create(context: Context, andesMessageAttrs: AndesBadgePillAttrs): AndesBadgeConfiguration {
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/badge/factory/AndesBadgeConfigurationFactory.kt:22]: cambiar por AndesBadgePillConfiguration, despues vamos a tener distintas sobrecargas del create para las configs del dot y el icon.

**Pull request merged**